src/domain/authentication/authService.ts

```diff
@@ -1,42 +1,79 @@
-import type { User, UserManagerSettings } from 'oidc-client';
-import { UserManager, Log, WebStorageStateStore } from 'oidc-client';
+import type { User, UserManagerSettings } from 'oidc-client-ts';
+import { UserManager, Log, WebStorageStateStore } from 'oidc-client-ts';
 import axios from 'axios';
 import * as Sentry from '@sentry/browser';
 
 import projectService from '../projects/projectService';
 import authorizationService from './authorizationService';
+import AppConfig from '../application/AppConfig';
 
 const origin = window.location.origin;
 export const API_TOKEN = 'apiToken';
 
+export type ApiTokenClientProps = {
+  url: string;
+  queryProps?: { permission: string; grant_type: string };
+  audiences?: string[];
+};
+
 export class AuthService {
   private userManager: UserManager;
+  private apiTokensClientConfig: ApiTokenClientProps;
+  private authServerType: 'KEYCLOAK' | 'TUNNISTAMO';
+  private audience: string;
 
   constructor() {
+    this.authServerType = AppConfig.oidcServerType ?? 'TUNNISTAMO';
+    this.audience = AppConfig.oidcAudience ?? AppConfig.oidcKukkuuApiClientId;
+
     const settings: UserManagerSettings = {
       loadUserInfo: true,
       userStore: new WebStorageStateStore({ store: window.localStorage }),
-      authority: process.env.REACT_APP_OIDC_AUTHORITY,
-      client_id: process.env.REACT_APP_OIDC_CLIENT_ID,
+      response_type: AppConfig.oidcReturnType,
+      authority: AppConfig.oidcAuthority,
+      client_id: AppConfig.oidcClientId,
+      scope: AppConfig.oidcScope,
       redirect_uri: `${origin}/callback`,
-      // For debugging, set it to 1 minute by removing comment:
-      // accessTokenExpiringNotificationTime: 59.65 * 60,
-      automaticSilentRenew: false,
-      silent_redirect_uri: `${origin}/silent_renew.html`,
-      response_type: 'id_token token',
-      scope: process.env.REACT_APP_OIDC_SCOPE,
       post_logout_redirect_uri: `${origin}/`,
+      // TODO: The silent renew support needs to be added to the React-admin authProvider as well.
+      // More about this:
+      // - https://marmelab.com/blog/2020/07/02/manage-your-jwt-react-admin-authentication-in-memory.html
+      // - https://marmelab.com/react-admin/addRefreshAuthToAuthProvider.html
+      // - https://marmelab.com/react-admin/addRefreshAuthToDataProvider.html
```

Comment:
👍 Good addition, now this makes it clearer that automatic silent renewal does not work.